modules/data.atmosphere/R/tdm_temporal_downscale_functions.R

```diff
@@ -0,0 +1,1127 @@
+##' @family tdm - Temporally Downscale Meteorology
+##' @param dat.train - training data generated by tdm_nc2dat.train.R
+##' @param n.beta - number of betas to generate
+##' @param path.out - path to where the training models & betas will be stored
+##' @param resids - whether or not to propogate residuals, set to FALSE
+##' @param parallel - whether or not to run in parallel. this is a feature
+##'                   still being worked on, set to FALSE
+##' @param n.cores - number of cores to use parallel processing on, set to NULL
+##' @param day.window - number of days surrounding current day we want to pull
+##'                     statistics from
+##' @param seed - allows this to be reproducible
+
+##' @author Christy Rollinson, James Simkins
+
+#----------------------------------------------------------------------
+# Begin Script
+#----------------------------------------------------------------------
+
+
+model.air_temperature <- function(dat.train, n.beta, path.out, day.window, 
+    resids = F, parallel = F, n.cores = NULL, seed = 1237) {
+    
+    set.seed(seed)
+    if (!dir.exists(path.out)) 
+        dir.create(path.out, recursive = T)
+    
+    # The model we're going to use
+    model.train <- function(dat.subset, n.beta, resids = resids) {
+        dat.subset$year <- as.ordered(dat.subset$year)
+
+        mod.doy <- lm(air_temperature ~ as.ordered(hour) * air_temperature_max.day * 
+            (lag.air_temperature + lag.air_temperature_min + air_temperature_min.day) + 
+            as.ordered(hour) * air_temperature_min.day * next.air_temperature_max - 
+            1 - as.ordered(hour) - lag.air_temperature - lag.air_temperature_min - 
+            next.air_temperature_max - air_temperature_max.day - air_temperature_min.day, 
+            data = dat.subset)  #
```

Comment:
There is a lot we could do to optimize code in PEcAn. To justify adding another dependency it would help to see if this is actually a bottleneck and more rigorously profile the downscaling workflow. Hadley's chapter on code profiling is very useful http://adv-r.had.co.nz/Profiling.html